### PR TITLE
Migration client 5.0.0 - 5.4.0 fixes

### PR DESCRIPTION
**Pull request by Saubhy**
- Migration client carries out all the DB queries on Identity DB. This was changed to use User management DB where necessary..
- Fixed Oracle DB scripts containing invalid characters.
- New local claims are added with each release and these are not being updated in the DB. This was fixed. 
- Fixed Oracle SQL scripts which had issues.
- Increased VARCHAR2(255) to 1023 in PASSWORD field of WS_BPS_PROFILE table.

**Note: DB script level fixes were done only for Oracle scripts.**

Github issue : https://github.com/wso2/product-is/issues/2575

### CODE REVIEW

**Comment by Saubhy** [comment]
Github Issue : https://github.com/wso2/product-is/issues/2575

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/migrator/ClaimsUpgrader.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
+/*
+* Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (thanujalk) [inline modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/migrator/ClaimsUpgrader.java]: Change the year

**Comment by ashensw** [comment]
@Saubhy please update the PR after resolving the conflicts.

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/migrator/ClaimsUpgrader.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+import org.wso2.carbon.is.migration.util.Utility;
+import org.wso2.carbon.user.api.Tenant;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.claim.inmemory.ClaimConfig;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.IOException;
+import java.util.*;
```
Comment (denuwanthi) [inline modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/migrator/ClaimsUpgrader.java]: Avoid * imports and import the specific classes

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/migrator/ClaimsUpgrader.java (hunk 1 above)
Comment (Saubhy) [reply modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/migrator/ClaimsUpgrader.java]: Will change

File: modules/migration/migration-resources/5.3.0/data/claim-config.xml [hunk 3]
```diff
@@ -0,0 +1,288 @@
+<!--
+ ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (Saubhy) [reply modules/migration/migration-resources/5.3.0/data/claim-config.xml]: Will change

File: modules/migration/migration-resources/5.3.0/data/claim-config.xml [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+            </Claim>
+            <Claim>
+                <ClaimURI>http://wso2.org/claims/active</ClaimURI>
+                <DisplayName>Active</DisplayName>
+                <AttributeID>active</AttributeID>
+                <Description>Status of the account</Description>
+            </Claim>
+        </Dialect>
+    </Dialects>
+</ClaimConfig>
```
[the remark this answers is not in the document]
Comment (Saubhy) [reply modules/migration/migration-resources/5.3.0/data/claim-config.xml]: Will do

File: modules/migration/migration-resources/5.4.0/data/claim-config.xml [hunk 5]
```diff
@@ -821,4 +543,4 @@
 			</Claim>
 		</Dialect>
 	</Dialects>
-</ClaimConfig>
+</ClaimConfig>
```
[the remark this answers is not in the document]
Comment (Saubhy) [reply modules/migration/migration-resources/5.4.0/data/claim-config.xml]: Will do

File: modules/migration/migration-resources/5.3.0/data/claim-config.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Saubhy) [reply modules/migration/migration-resources/5.3.0/data/claim-config.xml]: Will do

File: modules/migration/migration-resources/5.2.0/dbscripts/step1/um/oracle.sql [hunk 6]
```diff
@@ -105,4 +105,7 @@ CREATE INDEX REG_RESOURCE_IND_BY_UUID ON REG_RESOURCE(REG_UUID)
 CREATE INDEX REG_RESOURCE_IND_BY_TENAN ON REG_RESOURCE(REG_TENANT_ID, REG_UUID)
 /
 CREATE INDEX REG_RESOURCE_IND_BY_TYPE ON REG_RESOURCE(REG_TENANT_ID, REG_MEDIA_TYPE)
+/
+
+UPDATE UM_CLAIM SET UM_CLAIM_URI = 'urn:scim:schemas:core:1.0:locale' WHERE UM_CLAIM_URI = 'urn:scim:schemas:core:1.0:local'
 /
```
[the remark this answers is not in the document]
Comment (Saubhy) [reply modules/migration/migration-resources/5.2.0/dbscripts/step1/um/oracle.sql]: Will do

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/MigrationDatabaseCreator.java [hunk 7]
```diff
@@ -203,9 +197,12 @@ private void executeSQLScript(String dbscriptName) throws Exception {
                     }
                 }
                 //add the oracle database owner
-                if (!oracleUserChanged && "oracle".equals(databaseType) && line.contains("databasename :=")) {
-                    line = "databasename := '" + ISMigrationServiceDataHolder.getIdentityOracleUser() + "';";
-                    oracleUserChanged = true;
+                if ("oracle".equals(databaseType) && line.contains("databasename :=")) {
+                    if (dbscriptName.contains("identity")) {
```
[the remark this answers is not in the document]
Comment (Saubhy) [reply modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/MigrationDatabaseCreator.java]: Will do

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/MigrationDatabaseCreator.java [hunk 8]
```diff
@@ -203,9 +197,12 @@ private void executeSQLScript(String dbscriptName) throws Exception {
...
+                        line = "databasename := '" + ISMigrationServiceDataHolder.getIdentityOracleUser() + "';";
+                    } else if (dbscriptName.contains("um")) {
```
[the remark this answers is not in the document]
Comment (Saubhy) [reply modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/MigrationDatabaseCreator.java]: Will do

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/MigrationDatabaseCreator.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (Saubhy) [reply modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/MigrationDatabaseCreator.java]: This was existing code.

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/migrator/ClaimsUpgrader.java (hunk 2 above)
Comment (Saubhy) [reply modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/migrator/ClaimsUpgrader.java]: Will do

**Pull request merged**